pkg/syslLogger/loggers/loggerInterface.go

```diff
@@ -0,0 +1,22 @@
+package loggers
+
+type Logger interface {
+	Debug(args ...interface{})
+	Debugf(format string, args ...interface{})
+	Error(args ...interface{})
+	Errorf(format string, args ...interface{})
+	Exit(code int)
+	Fatal(args ...interface{})
+	Fatalf(format string, args ...interface{})
+	Panic(args ...interface{})
+	Panicf(format string, args ...interface{})
+	Print(args ...interface{})
+	Printf(format string, args ...interface{})
+	Trace(args ...interface{})
+	Tracef(format string, args ...interface{})
+	Warn(args ...interface{})
+	Warnf(format string, args ...interface{})
+	WithField(key string, val interface{}) Logger
```

Comment:
No, will change it to `PutField` and `PutFields` to avoid confusion

Reply:
No, leave it as `With...` and change the implementation to not affect the current logger please :)

Reply:
The thing is, `WithField` returns a logger so that it can chain fields insertion like this
```go
logger.WithField("val1", 1).WithField("val2", 2).WithField("val3", 3).Debug("This is a message")
```

If `WithField` copies the logger every time, it would be really slow. My plan was to just invoke `Copy` in the main API.

Is that better or returning a new `Logger` would be better?

Reply:
1. "it would be really slow" is a banned phrase.... PROVE IT!!
2. That is nonsensical code anyway, the caller should be doing `logger.WithFields(map[string]intereface{}{"val":1, "val2:2....`